### PR TITLE
implement bit-parallel GC counter

### PR DESCRIPTION
**Pull request by bicycle1885**
~Still an experimental implementation.~

This is a faster version of `gc_content` using a bit-parallel algorithm as I mentioned in https://github.com/BioJulia/BioSequences.jl/issues/27#issuecomment-342789976. I was inspired to try a bit-parallel algorithm by @timbitz, so I think you are interested in this pull request.

```
julia> srand(1234); seq = randdnaseq(1_000_000);

julia> @benchmark count(isGC, seq)
BenchmarkTools.Trial:
  memory estimate:  16 bytes
  allocs estimate:  1
  --------------
  minimum time:     7.788 ms (0.00% GC)
  median time:      8.220 ms (0.00% GC)
  mean time:        8.548 ms (0.00% GC)
  maximum time:     12.825 ms (0.00% GC)
  --------------
  samples:          585
  evals/sample:     1

julia> @benchmark BioSequences.count_gc(seq)
BenchmarkTools.Trial:
  memory estimate:  16 bytes
  allocs estimate:  1
  --------------
  minimum time:     171.186 μs (0.00% GC)
  median time:      171.251 μs (0.00% GC)
  mean time:        181.268 μs (0.00% GC)
  maximum time:     600.158 μs (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1

```

### CODE REVIEW

**Comment by bicycle1885** [comment]
I've added a two-bit version and some tests. I think this is in a good shape to review.

**Comment by timbitz** [comment]
Awesome @bicycle1885!  Sounds good, I'll take a look tomorrow morning.

File: src/sequence.jl [hunk 1]
```diff
@@ -107,30 +107,30 @@ end
...
-
     if isempty(seq)
         return 0.0
     else
-        return gc / length(seq)
+        return count_gc(seq) / length(seq)
     end
 end
 
+function count_gc(seq::Sequence)
```
Comment (timbitz) [inline src/sequence.jl]: Should this have the same name as count_gc in operators.jl?

Comment (timbitz) [reply src/sequence.jl]: Ohh i c. Its just for arbitrary other nucleic acid alphabets that don't fall into DNAAlphabet or RNAAlphabet... Cool.

Comment (bicycle1885) [reply src/sequence.jl]: That's right. Thank you for reviewing it.

Comment (timbitz) [reply src/sequence.jl]: Happy to review anytime!

**Pull request merged**